sklearn/_build_utils/check_openmp.pyx

```diff
@@ -0,0 +1,9 @@
+# the build will fail if OpenMP support is not handled correctly.
```

Comment:
Can you please explicitly describe the cases where this file fails?

Reply:
Actually I don't know exactly. I know one case of failure: building with a compiler which does not support openmp.

But with the added functionalities, either the build crashes before, with an explicit error message or one has set `SKLEARN_NO_OPENMP` and the build passes.

It's more like a unit test for the build. I'm not sure we should keep it. But there might be other reasons of failure that we don't know yet (there always are :/).

What explanation do you have in mind ?

Reply:
>What explanation do you have in mind ?

Well, I asked because I can't think of any reason to keep it either. It seems redundant with `check_openmp_support()`.

> there might be other reasons of failure that we don't know yet (there always are :/).

Definitely agree.  However, the fact that it's here makes me think there is a reason I don't understand, and it's making everything confusing ^^

I would suggest to remove it then, along with the changes to `test_configure()`.

Reply:
The function check_openmp is never called? Yes, I think if can be removed

Reply:
> I would suggest to remove it then, along with the changes to test_configure().

I removed it. However the changes to `test_configure()` are unrelated to that. Without those, the test would fail on osx with default apple clang compiler. We don't want to have to set all the env variables to enable openmp on default apple clang to run the test suite. One can use sklearn on a system which could not build scikit learn.

Reply:
>Without those, the test would fail on osx with default apple clang compiler.

Which tests would fail?

Reply:
`test_configure`, the one we are talking about :)

Reply:
So if I understand it correctly, the only use-case for this are mac users that:

- want to run the tests
- don't need to build scikit-learn (else they would have set the env variables already and the checks would work)

?

Reply:
not exactly because they would have set the env variable when building but it won't be set anymore when testing

Reply:
I see, thanks